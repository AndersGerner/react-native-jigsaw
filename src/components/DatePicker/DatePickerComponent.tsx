import React from "react";
import DateTimePicker from "@react-native-community/datetimepicker";
import { DatePickerComponentProps as Props } from "./DatePickerComponentType";

const DatePickerComponent: React.FC<Props> = ({
  value,
  onChange,
  mode,
  toggleVisibility,
}) => {
  return (
    <DateTimePicker
      value={value}
      mode={mode}
      onChange={(_event: any, data: any) => {
        toggleVisibility();
        onChange(null, data);
      }}
    />
  );
};

export default DatePickerComponent;
