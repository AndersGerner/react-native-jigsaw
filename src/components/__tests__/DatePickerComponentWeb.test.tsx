import * as React from "react";
import renderer from "react-test-renderer";
import DatePickerComponent from "../DatePicker/DatePickerComponent.web";

it("renders date picker component web", () => {
  const tree = renderer
    .create(
      <DatePickerComponent
        onChange={() => {}}
        mode="date"
        toggleVisibility={() => {}}
        value={new Date("01 Jan 1970 00:00:00 GMT")}
      />
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
