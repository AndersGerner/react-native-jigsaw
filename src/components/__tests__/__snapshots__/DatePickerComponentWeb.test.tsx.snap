// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders date picker component web 1`] = `null`;
